packages/api-generator/src/map.js

```diff
@@ -1,1378 +1,33 @@
-  { name: 'mustSort', source: 'v-data' },
-  { name: 'multiSort', source: 'v-data' },
-  { name: 'page', source: 'v-data' },
-  { name: 'itemsPerPage', source: 'v-data' },
-  { name: 'groupBy', source: 'v-data' },
-  { name: 'groupDesc', source: 'v-data' },
-  { name: 'locale', source: 'v-data' },
-  { name: 'disableSort', source: 'v-data' },
-  { name: 'disablePagination', source: 'v-data' },
-  { name: 'disableFiltering', source: 'v-data' },
-  { name: 'search', source: 'v-data' },
-  { name: 'customFilter', source: 'v-data', default: '(items: any[], search: string): any[]' },
-  { name: 'serverItemsLength', source: 'v-data' }
-]
-
-const dataIteratorSlots = [
-  { name: 'loading', source: 'data-iterator' },
-  { name: 'no-data', source: 'data-iterator' },
-  { name: 'no-results', source: 'data-iterator' }
-]
-
-const dataIteratorScopedSlots = [
-  {
-    name: 'default',
-    props: dataScopedProps,
-    source: 'data-iterator'
-  },
-  {
-    name: 'footer',
-    props: dataScopedProps,
-    source: 'data-iterator'
-  },
-  {
-    name: 'header',
-    props: dataScopedProps,
-    source: 'data-iterator'
-  }
-]
-
-const tableHeader = {
-  text: 'string',
-  value: 'string',
-  'align?': '\'start\' | \'center\' | \'end\'',
-  'sortable?': 'boolean',
-  'divider?': 'boolean',
-  'class?': 'string | string[]',
-  'width?': 'string | number',
-  'filter?': '(value: any, search: string, item: any): boolean',
-  'sort?': '(a: any, b: any): number'
-}
-
-const dataTableHeaderScopedProps = {
-  props: {
-    options: dataOptions,
-    pagination: dataPagination
-    // TOOD: Also the rest of v-data-footer props
-  },
-  on: {
-    'update:options': '(value: any): void'
-  },
-  widths: 'number[]',
-  headers: 'TableHeader[]' // TODO: expand this?
-}
-
-const dataTableSlots = [
-  { name: 'body.append', props: dataScopedProps },
-  { name: 'body.prepend', props: dataScopedProps },
-  { name: 'body', props: dataScopedProps },
-  { name: 'footer', props: dataScopedProps },
-  { name: 'header', props: dataTableHeaderScopedProps },
-  { name: 'top', props: dataScopedProps },
-  { name: 'progress', props: dataScopedProps },
-  { name: 'group', props: dataScopedProps },
-  { name: 'group.header', props: dataScopedProps },
-  { name: 'group.summary', props: dataScopedProps },
-  { name: 'item', props: dataScopedProps },
-  { name: 'item.dataTableSelect', props: dataScopedProps },
-  { name: 'item.dataTableExpand', props: dataScopedProps },
-  { name: 'item.column.<name>', props: dataScopedProps },
-  { name: 'item.expanded', props: dataScopedProps }
-]
-
-const dataFooterSlots = [
-  { name: 'pageText' }
-]
+const fs = require('fs')
 
-const dataFooterEvents = [
-  { name: 'update:options' }
-]
-
-const validatableEvents = [
-  {
-    name: 'update:error',
-    value: 'boolean'
-  }
-]
+// get datatable data
+const vData = require('./js/v-data')
```

Comment:
Why are these separate from all other files? Hardcoding this is _not_ recommended

Reply:
Converted everything to JS.